assets/pages/regular-contributions/helpers/ajax.js

```diff
@@ -20,25 +23,50 @@ const MAX_POLLS = 10;
 
 // ----- Types ----- //
 
-type RegularContribFields = {
-  contribution: {
-    amount: number,
-    currency: string,
-    billingPeriod: BillingPeriod,
-  },
+type ContributionRequest = {
+  amount: number,
+  currency: string,
+  billingPeriod: BillingPeriod,
+};
+
+type RegularContribFields = {|
+  firstName: ?string,
+  lastName: ?string,
+  country: IsoCountry,
+  state?: UsState,
+  contribution: ContributionRequest,
   paymentFields: {
     stripeToken: string,
   },
-  state?: UsState,
-  firstName: ?string,
-  lastName: ?string,
-};
+  ophanIds: OphanIds,
+  referrerAcquisitionData: ReferrerAcquisitionData,
+  supportAbTests: AcquisitionABTest[],
+|};
 
 type PaymentField = 'baid' | 'stripeToken';
 
 
 // ----- Functions ----- //
 
+const isStateValid = (inputState: Object) => {
+  const user = inputState.user;
+
+  return user.firstName !== null && user.firstName !== undefined &&
+         user.lastName !== null && user.lastName !== undefined &&
+         user.email !== null && user.email !== undefined;
+};
+
+function removeNullFields(obj: Object): Object {
+
+  const response = obj;
```

Comment:
Isn't this copy by reference? You're modifying the original object here as well, no?